source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc

```diff
@@ -136,6 +143,51 @@ bool LocalRateLimiterImpl::requestAllowed(
   return requestAllowedHelper(tokens_);
 }
 
+uint32_t LocalRateLimiterImpl::maxTokens(
+    absl::Span<const RateLimit::LocalDescriptor> request_descriptors) const {
+  if (!descriptors_.empty() && !request_descriptors.empty()) {
+    for (const auto& request_descriptor : request_descriptors) {
+      auto it = descriptors_.find(request_descriptor);
+      if (it != descriptors_.end()) {
+        return it->token_bucket_.max_tokens_;
+      }
+    }
+  }
+  return token_bucket_.max_tokens_;
+}
```

Comment:
Should this be finding the max of all the descriptors? Seems like it's finding the max of the first descriptor we know about? 

Docs might be good here if this is the intended implementation, I'm not that familiar with this code and seeing the impl for a function called `maxTokens` is a bit confusing

Reply:
No, each request will correspond to a certain descriptor, and these descriptors have their own token bucket. So what we need to do here is to find the corresponding token bucket and return its `macTokens`.

Reply:
Thanks for the explanation. Just for my own learning purpose: 
You are passing a set of request_descriptors into this function and iterate on it. Is it possible that we can find multiple entries from `request_descriptors` in `descriptors_`. i.e., `it != descriptors_.end()` happens multiple times in the for loop.

My previous comment about max_token was referring to such case. But I could be just wrong as I am not that familiar with code here.

Reply:
NVM. I found the answer:)
From the doc:  The override rate limit descriptor is selected by the *first full match* from the request descriptors.
So this pattern is intended behavior. 

It might be a good idea to add this comment to the code? in case other people are wondering this as well.

Reply:
Thanks, I will add the comment.